health-checker/src/main/java/io/stargate/health/DataStoreHealthChecker.java

```diff
@@ -82,4 +102,45 @@ protected Result check() {
       return Result.unhealthy("Unable to access DataStore: " + e);
     }
   }
+
+  private void ensureTableExists(DataStore dataStore)
+      throws ExecutionException, InterruptedException {
+    // create a dedicated health-check keyspace to not risk conflicts with users keyspace
+    dataStore
+        .execute(
+            dataStore
+                .queryBuilder()
+                .create()
+                .keyspace(KEYSPACE_NAME)
+                .ifNotExists()
+                .withReplication(Replication.simpleStrategy(1))
+                .build()
+                .bind())
+        .get();
+
+    // create table
+    dataStore
+        .execute(
+            dataStore
+                .queryBuilder()
+                .create()
+                .table(KEYSPACE_NAME, TABLE_NAME)
+                .ifNotExists()
+                .column(EXPECTED_TABLE.columns())
+                .build()
+                .bind())
+        .get();
```

Comment:
These should be in a static block to only run once. If at some point the keyspace and/or table disappear then that's a problem.

Reply:
Right, constructor. If it fails on init then we want that to fail startup because something is not in a healthy state.